lib/symboldatabase.cpp

```diff
@@ -3920,6 +3920,28 @@ void SymbolDatabase::printVariable(const Variable *var, const char *indent) cons
     std::cout << std::endl;
 }
 
+static std::string functionTypeToString(Function::Type type)
```

Comment:
Returning `const char*` would preserve the previous behavior and would also avoid generate an temporary `std::string` (not sure if inlining would get rid of that).

Reply:
Thing is, there are similar helper functions which return `std::string`. Adjusted and grouped together.

Reply:
I also see use of `std::endl` and `std::ostringstream`, which probably doesn't help performance either. Or it might just be irrelevant.

Reply:
> Thing is, there are similar helper functions (`scopeTypeToString`, `accessControlToString`) which return `std::string`. Adjusted and grouped together.

It shouldn't really matter much. Just mentioned it because it changed the behavior.

> I also see use of `std::endl` and `std::ostringstream`, which probably doesn't help performance either. Or it might just be irrelevant.

The most relevant places have already been adjusted. I am not going into profiling again until my other big tasks are finally done and I filed some more upstream tickets about suboptimal code generation.